npm/private/npm_translate_lock_helpers.bzl

```diff
@@ -564,6 +564,27 @@ def _normalize_bazelignore(lines):
             result.append(line.rstrip(strip_trailing_chars))
     return result
 
+################################################################################
+def _verify_lifecycle_hooks_specified(_, state):
+    # lockfiles <9.0 specify the `requiresBuild` flag in the lockfile.
+    #
+    # lockfiles >=9.0 no longer specify if packages have lifecycle hooks,
+    # and declaration of hooks must be done manually in the `pnpm.onlyBuiltDependencies`.
+    if state.lockfile_version() < 9.0:
+        return
+
+    if state.only_built_dependencies() == None:
+        fail("""\
+ERROR: in verify_lifecycle_hooks_specified:
+
+As of pnpm v9 the lockfile no longer specifies if packages have lifecycle hooks.
```

Comment:
Lets add a little some more "why" and a link to that pnpm issue re: requiresBuild to this error message. The more verbose the better here as this is part of the upgrade documentation.

```
As of pnpm v9, the lockfile no longer specifies if packages have lifecycle hooks.

Packages that rules_js should generate lifecycle hook actions for must now be declared in pnpm.onlyBuiltDependencies in the pnpm workspace root package.json. See https://pnpm.io/package_json#pnpmonlybuiltdependencies for more information.

Prior to pnpm v9, rules_js keyed off of the requiresBuild attribute in the pnpm lock file to determine if a lifecycle hook action should be generated for an npm package. See <link> for the reasons that pnpm removed the requiredBuild attribute from the lockfile in v9.
```

Reply:
When we add the "strict" check we can mention that in this error message as well.